es/es-server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -219,17 +220,29 @@ protected void setupFieldType(BuilderContext context) {
                 fieldType.setHasDocValues(defaultDocValues);
             }
         }
+
+        public void position(int position) {
```

Comment:
Why is it a primitive here?

Reply:
To enforce the non-null. This method either shouldn't be used, in which case the default (`null`) is used, or if it is used it must have a value.